packages/react-dom/src/__tests__/ReactBrowserEventEmitter-test.internal.js

```diff
@@ -9,360 +9,480 @@
 
     container = document.createElement('div');
     document.body.appendChild(container);
 
-    let GRANDPARENT_PROPS = {};
-    let PARENT_PROPS = {};
-    let CHILD_PROPS = {};
-
-    function Child(props) {
-      return <div ref={c => (CHILD = c)} {...props} />;
+    class Child extends React.PureComponent {
+      render() {
+        return <div {...this.props} />;
+      }
     }
 
-    class ChildWrapper extends React.PureComponent {
+    class Parent extends React.Component {
       render() {
-        return <Child {...this.props} />;
+        return (
+          <div {...this.props.parentProps}>
+            <Child ref={n => (childRef = n)} {...this.props.childProps} />
+          </div>
+        );
       }
     }
 
-    function renderTree() {
-      ReactDOM.render(
-        <div ref={c => (GRANDPARENT = c)} {...GRANDPARENT_PROPS}>
-          <div ref={c => (PARENT = c)} {...PARENT_PROPS}>
-            <ChildWrapper {...CHILD_PROPS} />
-          </div>
-        </div>,
+    createInitialInstance = () => {
+      const initialInstance = ReactDOM.render(
+        <Parent parentProps={{}} childProps={{onClick: LISTENER}} />,
         container,
       );
-    }
+      expect(childRef.props.onClick).toBe(LISTENER);
 
-    renderTree();
-
-    getListener = function(node, eventName) {
-      const inst = ReactDOMComponentTree.getInstanceFromNode(node);
-      return EventPluginHub.getListener(inst, eventName);
-    };
-    putListener = function(node, eventName, listener) {
-      switch (node) {
-        case CHILD:
-          CHILD_PROPS[eventName] = listener;
-          break;
-        case PARENT:
-          PARENT_PROPS[eventName] = listener;
-          break;
-        case GRANDPARENT:
-          GRANDPARENT_PROPS[eventName] = listener;
-          break;
-      }
-      // Rerender with new event listeners
-      renderTree();
-    };
-    deleteAllListeners = function(node) {
-      switch (node) {
-        case CHILD:
-          CHILD_PROPS = {};
-          break;
-        case PARENT:
-          PARENT_PROPS = {};
-          break;
-        case GRANDPARENT:
-          GRANDPARENT_PROPS = {};
-          break;
-      }
-      renderTree();
+      return initialInstance;
     };
 
-    idCallOrder = [];
+    updateInstance = ({parentProps, childProps}) => {
+      return ReactDOM.render(
+        <Parent parentProps={parentProps} childProps={childProps} />,
+        container,
+      );
+    };
   });
 
   afterEach(() => {
     document.body.removeChild(container);
     container = null;
+    childRef = null;
   });
 
   it('should store a listener correctly', () => {
-    registerSimpleTestHandler();
-    const listener = getListener(CHILD, ON_CLICK_KEY);
-    expect(listener).toBe(LISTENER);
+    createInitialInstance();
+    updateInstance({
+      parentProps: {},
+      childProps: {
+        onClick: LISTENER,
+      },
+    });
+    expect(childRef.props.onClick).toBe(LISTENER);
```

Comment:
This test is not asserting the same thing as the previous one. The previous test used to call the internal API `ReactDOMComponentTree.getInstanceFromNode(node);` to find out if React internally added an event listener. The new code is asserting that the onClick prop is set which is what you do in the line above. I think this would be green even if we remove the event system.

I think this test can safely be removed though. We have _a lot_ of coverage for simple `onClick` cases.